### ISSUE TITLE
How to run while(true) of message consuming  in sub section into the different thread ?

### ISSUE THREAD

**Issue by BlackSwallow96** [sewenew/redis-plus-plus#500]
hi guys , i am a newbie developer ... sorry about my question if not a good question ...
in default example of subscriber we have while (true) statement which lock excution of program . i want to run sub.consum() in different thread like this :

std::atomic\<bool\> end_of_program = false;

int main()
{
 // conecting and initializing code goes here
  thread t(cosuming, &sub);
  t.detach();
// some code goes here

    end_of_program = true;
}
void consuming(Subscriber *sub)
{
     while(!end_of_program)
    {
          sub.consum();
    }
}

i this situation my excution of my code not blocked by while true statement . but i get some errors and exceptions 
what i have to do ?

**Comment by sewenew** [comment]
You need to set socket_timeout for connection option, and join worker thread in main thread. Check [this](https://github.com/sewenew/redis-plus-plus/issues/171) for an example.

Regards

**Comment by BlackSwallow96** [comment]
yes worked for me ... 
thank you very much 
best Regards
